src/Task.cc

```diff
@@ -3210,8 +3210,10 @@ static ssize_t safe_pwrite64(Task* t, const void* buf, ssize_t buf_size,
       // We could be trying to write to a read-only shared file. In that case we should
       // report the error without dying.
       failed_access = true;
-    } else {
-      remote.check_syscall_result(ret, mprotect_syscallno, false);
+    } else if (remote.check_syscall_result(ret, mprotect_syscallno)) {
```

Comment:
Note `false` as last parameter to `check_syscall_result` in the removed  code. It makes `safe_pwrite64` infallible during record which is not what we want.

Reply:
@rocallahan Sorry for the ping -- the PR may not have been very clear when I posted it originally. 

My subsequently added comment (above) regarding passing `false` for `bool allow_death` might make things clearer why this PR is necessary. 

`allow_death` should be `true` during record -- and this the default parameter value.